core/Symbols.cc

```diff
@@ -1215,9 +1215,11 @@ void Symbol::addLoc(const core::GlobalState &gs, core::Loc loc) {
     }
 
     if (loc.file().data(gs).sourceType == core::File::Type::Normal && !loc.file().data(gs).isRBI()) {
-        locs_.insert(locs_.begin(), loc);
-    } else {
+        // Make this the new canonical loc.
         locs_.emplace_back(loc);
+    } else {
+        // If we have other locs, use them as the canonical loc.
+        locs_.insert(locs_.begin(), loc);
```

Comment:
what if you insert this one before the end, instead of at the start?

then it preserves the existing order of the locs (that newer locs come later in the list

Reply:
By that you mean at index `len - 2`?

Reply:
(Note that this is only for RBI locs.)